pkgs/development/haskell-modules/configuration-ghc-8.0.x.nix

```diff
@@ -38,6 +38,13 @@ self: super: {
   # cabal-install can use the native Cabal library.
   cabal-install = super.cabal-install.override { Cabal = null; };
 
+  idris = overrideCabal super.idris (drv: {
+    # "idris" binary cannot find Idris library otherwise while building.
+    # After installing it's completely fine though. Nix-specific issue so not
+    # reported.
+    preBuild = "export LD_LIBRARY_PATH=$PWD/dist/build:$LD_LIBRARY_PATH";
```

Comment:
> when the build gets to the part where the idris binary will be invoked to start actually compiling the Idris stdlib, the build fails, since ld.so can't find the library against which idris is linked.

These build processes are driven entirely by Cabal. I don't see how Nix could cause that error.

> the default is to build a shared library, AFAIK

Maybe so, but that doesn't mean that executables are linked against it. Note that the default `~/.cabal/config` file says `executable-dynamic: False`.

>  is there a problem with fixing it for ghc 8.0.x as well?

No, I am totally in favor of the fix. I'd just like to see this issue reported upstream, too, so that the developers are aware of the problem and have a chance to fix it. Ideally, the override would refer to an appropriate ticket URL, so that other Nixpkgs maintainers can easily find out what is going on and why that override was added.